src/poetry/repositories/pypi_repository.py

```diff
@@ -147,7 +147,7 @@ def find_links_for_package(self, package: Package) -> list[Link]:
 
     def _get_release_info(
         self, name: NormalizedName, version: Version
-    ) -> dict[str, str | list[str] | None]:
```

Comment:
was this type that bad? I feel that `Any` is always worse choice over more concrete type

Reply:
this one was definitely wrong iirc, there was some point at which one of the values was a list of dictionaries or something like that.

Try changing it back and see what additional warning comes up if you really want to know!

Reply:
https://github.com/python-poetry/poetry/blob/978d435be25b799fb6d107c1fb1739eb858057c1/tests/repositories/test_pypi_repository.py#L353-L359 is the testcase that reveals the old type to be a lie